### PR TITLE
boards: shields: Add X-NUCLEO-IDB05A1

### PR DESCRIPTION
**Pull request by yboutreux**
Add STMicroelectronics BLE shield X-NUCLEO-IDB05A1.
Add configurations for using X-NUCLEO-IDB05A1 with stm32mp157c_dk2 board.

### CODE REVIEW

File: boards/shields/x_nucleo_idb05a1/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
+.. _x-nucleo-idb05a1:
+
+X-NUCLEO-IDB05A1: Bluetooth Low Energy expansion board based on the SPBTLE-RF
+module
+#####################################################################
```
Comment (erwango) [inline boards/shields/x_nucleo_idb05a1/doc/index.rst]: I guess this title syntax should be fixed

**Review by erwango — APPROVED** [review]
Approved, but @dbkinder might have more comments on doc

File: boards/shields/x_nucleo_idb05a1/doc/index.rst (hunk 1 above)
Comment (yboutreux) [reply boards/shields/x_nucleo_idb05a1/doc/index.rst]: If you are refering to "SPBTLE-RF" and "module" words being split on two lines, this is done to comply with "80 character max per line" rule

Comment (erwango) [reply boards/shields/x_nucleo_idb05a1/doc/index.rst]: Yes, but I think this won't work with rst or html syntax

Comment (yboutreux) [reply boards/shields/x_nucleo_idb05a1/doc/index.rst]: Tried with rst2pdf, indeed this break the title, nice catch thank you !

**Comment by galak** [comment]
@dbkinder can you review the docs.

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: boards/shields/x_nucleo_idb05a1/x_nucleo_idb05a1.overlay [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&arduino_spi {
+	cs-gpios = <&arduino_header 16 0>;		/* D10 */
+
+	spbtle-rf@0 {
+		compatible = "zephyr,bt-hci-spi";
+		reg = <0>;
+		reset-gpios = <&arduino_header 13 0>;	/* D7 */
```
Comment (erwango) [inline boards/shields/x_nucleo_idb05a1/x_nucleo_idb05a1.overlay]: This is not the default configuration, this should be moved to A1

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/shields/x_nucleo_idb05a1/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+The SPBTLE-RF module is FCC (FCC ID: S9NSPBTLERF) and IC certified
+(IC: 8976C-SPBTLERF).
+
+The X-NUCLEO-IDB05A1 is compatible with the ST Morpho and Arduino UNO R3
+connector layout (the user can mount the ST Morpho connectors, if required). The
+X-NUCLEO-IDB05A1 interfaces with the host microcontroller via the SPI pin, and
+the user can change the default SPI clock, the SPI chip select and SPI IRQ by
+changing one resistor on the evaluation board.
+
+IMPORTANT NOTICE : Some modification must be done out of the box, please refer
```
[the remark this answers is not in the document]
Comment (erwango) [reply boards/shields/x_nucleo_idb05a1/doc/index.rst]: @dbkinder, I think this point is indeed important and should be somehow highlighted.
Shield provides a pseudo arduino compatible pinnout, close enough to oversight the a documentation detail and far enough to loose few hours of debugging.

File: boards/shields/x_nucleo_idb05a1/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+     :height: 350px
+     :align: center
+     :alt: X-NUCLEO-IDB05A1
+
+More information about the board can be found at the
+`X-NUCLEO-IDB05A1 website`_.
+
+Hardware configuration
+**********************
+To follow Arduino pinout, resistors R2 and R4 must be removed and please connect
```
[the remark this answers is not in the document]
Comment (erwango) [reply boards/shields/x_nucleo_idb05a1/doc/index.rst]: @dbkinder, `Depending on your host board`, well this is most of (all ?) boards that declare arduino compatible header, at least for the R4/R6 change.

**Comment by erwango** [comment]
Replaced by #19277

**Pull request closed without merging**